### PR TITLE
Public xtest preparation

### PR DESCRIPTION
**Pull request by jbech-linaro**
Contains updates for FVP and QMEU to use the public optee_test (which also are using ta-dev-kit built by optee_os).

### CODE REVIEW

File: scripts/setup_qemu_optee.sh [hunk 1]
```diff
@@ -462,26 +454,31 @@ cd $DEV_PATH
...
-  echo "CFG_GP_PACKAGE_PATH must be the path to the GP testsuite directory"
-  exit 1
-fi
-make patch
+	export CFG_GP_PACKAGE_PATH=\${CFG_GP_PACKAGE_PATH:-$DST_OPTEE_TEST/TEE_Initial_Configuration-Test_Suite_v1_1_0_4-2014_11_07}
+	if [ ! -d "\$CFG_GP_PACKAGE_PATH" ]; then
+		echo "CFG_GP_PACKAGE_PATH must be the path to the GP testsuite directory"
+		exit 1
+	fi
+	make patch
```
Comment (pascal-brand-st-dev) [inline scripts/setup_qemu_optee.sh]: Add TA_DEV_KIT_DIR definition:

    make patch TA_DEV_KIT_DIR=$DEV_PATH/optee_os/out/arm-plat-vexpress/export-user_ta

File: scripts/setup_qemu_optee.sh [hunk 2]
```diff
@@ -462,26 +454,31 @@ cd $DEV_PATH
 cat > $DEV_PATH/build_optee_tests.sh << EOF
 #!/bin/bash
 cd $DST_OPTEE_TEST
-export PATH=$DST_AARCH32_GCC/bin:\$PATH
 
 export CFG_DEV_PATH=$DEV_PATH
 export CFG_PLATFORM_FLAVOR=qemu_virt
-export CFG_ROOTFS_DIR=\$CFG_DEV_PATH/out
+export CFG_ROOTFS_DIR=\$CFG_DEV_PATH/out/optee_test
+
```
Comment (pascal-brand-st-dev) [inline scripts/setup_qemu_optee.sh]: Missing the following when using GP test:

    export CFG_ARM32=y

File: scripts/setup_qemu_optee.sh [hunk 3]
```diff
@@ -462,26 +454,31 @@ cd $DEV_PATH
 cat > $DEV_PATH/build_optee_tests.sh << EOF
 #!/bin/bash
 cd $DST_OPTEE_TEST
-export PATH=$DST_AARCH32_GCC/bin:\$PATH
 
 export CFG_DEV_PATH=$DEV_PATH
 export CFG_PLATFORM_FLAVOR=qemu_virt
-export CFG_ROOTFS_DIR=\$CFG_DEV_PATH/out
+export CFG_ROOTFS_DIR=\$CFG_DEV_PATH/out/optee_test
```
Comment (pascal-brand-st-dev) [inline scripts/setup_qemu_optee.sh]: Missing

    export CFG_OPTEE_TEST_PATH=$CFG_DEV_PATH/optee_test

**Comment by pascal-brand-st-dev** [comment]
When the setup qemu has been updated with 3 above comments:

    Tested-by: Pascal Brand <pascal.brand@linaro.org> (QEMU) (standard and extended tests)

**Comment by jbech-linaro** [comment]
@pascal-brand-st-dev what about FVP? Same changes are needed there I guess? What should CFG_ARM32 be in that case? (FVP is 64-bit host, but we're still defaulting to 32-bit on secure side).

**Comment by pascal-brand-st-dev** [comment]
On FVP, I did not get issues related to CFG_ARM32. I did not have to add it in the scripts.

    Tested-by: Pascal Brand <pascal.brand@linaro.org> (FVP platform - standard tests)

**Comment by jbech-linaro** [comment]
Tags applied, comments addressed and rebased. Ready to merge.

**Comment by pascal-brand-st-dev** [comment]
Sorry for the new comments. I though they were sent  :(

**Comment by jbech-linaro** [comment]
@pascal-brand-st-dev no problems. I actually had it fixes, but when you said in previous comments earlier today that you hadn't seen any problems with CFG_ARM32 I took for granted that the other lines weren't necessary either and I simply removed that :-) I'll add them again. No problem.

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by pascal-brand-st** [comment]
Merged manually.

**Pull request closed without merging**